packages/router-api/src/primitives.rs

```diff
@@ -41,11 +41,18 @@ pub struct Message {
 impl Message {
     pub fn hash(&self) -> Hash {
         let mut hasher = Keccak256::new();
+        let delimiter_bytes = &[CHAIN_NAME_DELIMITER as u8];
+
         hasher.update(self.cc_id.to_string());
+        hasher.update(delimiter_bytes);
         hasher.update(self.source_address.as_str());
```

Comment:
prevent delimiter to occur in other fields as well

Reply:
You can check for this in the address type, pretty sure all of the other fields disallow _  already